tremor-pipeline/src/op/qos/wal.rs

```diff
@@ -184,47 +196,60 @@ op!(WalFactory(_uid, node) {
     if let Some(map) = &node.config {
         let config: Config = Config::new(map)?;
 
-        Ok(Box::new(Wal::new(node.id.to_string(), config)?))
+        if let (None, None) = (config.max_elements, config.max_bytes) {
+            Err(ErrorKind::BadOpConfig("WAL operator needs at least one of `max_elements` or `max_bytes` config entries.".to_owned()).into())
+        } else {
+            Ok(Box::new(Wal::new(node.id.to_string(), config)?))
+        }
 
     } else {
         Err(ErrorKind::MissingOpConfig(node.id.to_string()).into())
     }
 });
 
 impl Wal {
+    const URI_SCHEME: &'static str = "tremor-wal";
+    const READ: &'static str = "read";
```

Comment:
while this is changing, should we rename this as 'confirmed' ?

Reply:
i dont think so, it is used for the pointer to the event we have already read and emitted downstream, no?

Reply:
ah, now i see what you mean, yeah makes sense